module/zfs/arc.c

```diff
@@ -10442,7 +10445,8 @@ l2arc_log_blk_commit(l2arc_dev_t *dev, zio_t *pio, l2arc_write_callback_t *cb)
 
 	/* try to compress the buffer */
 	psize = zio_compress_data(ZIO_COMPRESS_LZ4,
-	    abd_buf->abd, (void **) &tmpbuf, sizeof (*lb), 0);
+	    abd_buf->abd, (void **) &tmpbuf, sizeof (*lb), 0,
+	    1 << dev->l2ad_vdev->vdev_ashift);
```

Comment:
This also needs to be updated to the zio_get_compression_max_size() logic, just with the 1<<ashift instead of spa values.